<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>ElevatorSimQt - v0.6.1 (Single-Select + Dual Logs + Hybrid)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,1">
    <!-- Connection -->
    <item>
     <widget class="QGroupBox" name="grpConn">
      <property name="title"><string>Connection</string></property>
      <layout class="QGridLayout" name="gridConn">
       <item row="0" column="0">
        <widget class="QLabel" name="lblPort">
         <property name="text"><string>Port</string></property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QComboBox" name="cmbPorts"/>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="btnRefresh">
         <property name="text"><string>Refresh</string></property>
        </widget>
       </item>

       <item row="1" column="0">
        <widget class="QLabel" name="lblBaud">
         <property name="text"><string>Baud</string></property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="spinBaud">
         <property name="minimum"><number>1200</number></property>
         <property name="maximum"><number>10000000</number></property>
         <property name="value"><number>115200</number></property>
         <property name="singleStep"><number>1200</number></property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="btnConnect">
         <property name="text"><string>Connect</string></property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="btnDisconnect">
         <property name="text"><string>Disconnect</string></property>
         <property name="enabled"><bool>false</bool></property>
        </widget>
       </item>
       <!-- Optional toggles (hybrid anim always on; these can be ignored) -->
       <item row="1" column="4">
        <widget class="QCheckBox" name="chkAnimate">
         <property name="text"><string>Animate UI</string></property>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QCheckBox" name="chkSim">
         <property name="text"><string>Simulator mode</string></property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Controls -->
    <item>
     <widget class="QGroupBox" name="grpControls">
      <property name="title"><string>Controls</string></property>
      <layout class="QGridLayout" name="gridControls">
       <!-- Floors row -->
       <item row="0" column="0">
        <widget class="QLabel" name="lblFloors">
         <property name="text"><string>Floors</string></property>
        </widget>
       </item>
       <item row="0" column="1" colspan="5">
        <layout class="QHBoxLayout" name="hboxFloors">
         <item><widget class="QPushButton" name="btnF1"><property name="text"><string>1</string></property></widget></item>
         <item><widget class="QPushButton" name="btnF2"><property name="text"><string>2</string></property></widget></item>
         <item><widget class="QPushButton" name="btnF3"><property name="text"><string>3</string></property></widget></item>
         <item><widget class="QPushButton" name="btnF4"><property name="text"><string>4</string></property></widget></item>
         <item><widget class="QPushButton" name="btnF5"><property name="text"><string>5</string></property></widget></item>
        </layout>
       </item>
       <!-- Door + status request -->
       <item row="1" column="0">
        <widget class="QLabel" name="lblDoorCtl">
         <property name="text"><string>Door</string></property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btnDoorOpen">
         <property name="text"><string>Open</string></property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="btnDoorClose">
         <property name="text"><string>Close</string></property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="btnStatus">
         <property name="text"><string>Status?</string></property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Status -->
    <item>
     <widget class="QGroupBox" name="grpStatus">
      <property name="title"><string>Status</string></property>
      <layout class="QGridLayout" name="gridStatus">
       <item row="0" column="0"><widget class="QLabel" name="lblCurFloor"><property name="text"><string>Floor:</string></property></widget></item>
       <item row="0" column="1"><widget class="QLabel" name="valCurFloor"><property name="text"><string>-</string></property></widget></item>
       <item row="0" column="2"><widget class="QLabel" name="lblMoving"><property name="text"><string>Moving:</string></property></widget></item>
       <item row="0" column="3"><widget class="QLabel" name="valMoving"><property name="text"><string>IDLE</string></property></widget></item>
       <item row="1" column="0"><widget class="QLabel" name="lblDoor"><property name="text"><string>Door:</string></property></widget></item>
       <item row="1" column="1"><widget class="QLabel" name="valDoor"><property name="text"><string>CLOSED</string></property></widget></item>
      </layout>
     </widget>
    </item>

    <!-- Logs -->
    <item>
     <widget class="QGroupBox" name="grpLog">
      <property name="title"><string>Logs</string></property>
      <layout class="QGridLayout" name="gridLogs">
       <item row="0" column="0">
        <widget class="QLabel" name="lblConnLog">
         <property name="text"><string>Device Log</string></property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPlainTextEdit" name="textConnLog">
         <property name="readOnly"><bool>true</bool></property>
         <property name="minimumHeight"><number>90</number></property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="lblFloorLog">
         <property name="text"><string>Floor Log</string></property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPlainTextEdit" name="textFloorLog">
         <property name="readOnly"><bool>true</bool></property>
         <property name="minimumHeight"><number>70</number></property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
